packages/protocol/contracts/governance/Proposals.sol

```diff
@@ -0,0 +1,351 @@
+   * @return The participation of the proposal.
+   */
+  function _getParticipation(Proposal storage proposal) internal view returns (FixidityLib.Fraction memory) {
+    uint256 totalVotes = proposal.votes.yes.add(proposal.votes.no).add(proposal.votes.abstain);
+    return FixidityLib.newFixed(totalVotes).divide(FixidityLib.newFixed(proposal.networkWeight));
+  }
+
+  /**
+   * @notice Returns a specified transaction in a proposal.
+   * @param proposal The proposal struct.
+   * @param index The index of the specified transaction in the proposal's transaction list.
+   * @return The specified transaction.
+   */
+  function getTransaction(
+    Proposal storage proposal,
+    uint256 index
+  )
+    public
+    view
+    returns (uint256, address, bytes memory)
+  {
+    require(index < proposal.transactions.length);
+    Transaction storage transaction = proposal.transactions[index];
+    return (transaction.value, transaction.destination, transaction.data);
+  }
+
+  /**
+   * @notice Returns an unpacked proposal struct with its transaction count.
+   * @param proposal The proposal struct.
+   * @return The unpacked proposal with its transaction count.
+   */
+  function unpack(
+    Proposal storage proposal
+  )
+    internal
+    view
+    returns (address, uint256, uint256, uint256)
+  {
+    return (
+      proposal.proposer,
+      proposal.deposit,
+      proposal.timestamp,
+      proposal.transactions.length
+    );
+  }
+
+  /**
+   * @notice Returns the referendum vote totals for a proposal.
+   * @param proposal The proposal struct.
+   * @return The yes, no, and abstain vote totals.
+   */
+  function getVoteTotals(
+    Proposal storage proposal
+  )
+    internal
+    view
+    returns (uint256, uint256, uint256)
+  {
+    return (proposal.votes.yes, proposal.votes.no, proposal.votes.abstain);
+  }
+
+  /**
+   * @notice Returns whether or not a proposal has been approved.
+   * @param proposal The proposal struct.
+   * @return Whether or not the proposal has been approved.
+   */
+  function isApproved(Proposal storage proposal) internal view returns (bool) {
+    return proposal.approved;
+  }
+
+  /**
+   * @notice Returns whether or not a proposal exists.
+   * @param proposal The proposal struct.
+   * @return Whether or not the proposal exists.
+   */
+  function exists(Proposal storage proposal) internal view returns (bool) {
+    return proposal.timestamp > 0;
+  }
+
+  // call has been separated into its own function in order to take advantage
+  // of the Solidity's code generator to produce a loop that copies tx.data into memory.
+  /**
+   * @notice Executes a function call.
+   * @param value The value of Celo Gold to be sent with the function call.
+   * @param destination The destination address of the function call.
+   * @param dataLength The length of the data to be included in the function call.
+   * @param data The data to be included in the function call.
+   */
+  function externalCall(
```

Comment:
Note that there's an outstanding issue to check that the destination address exists